### ISSUE TITLE
Appium Server OOM during loop screenshot

### ISSUE THREAD

**Issue by lanbaoshen** [appium/appium#20451]
### Do I have the most recent component updates?

- [X] I use the most recent available driver/plugin and server versions

### Is the component officially supported by the Appium team?

- [X] I have verified the component repository is present under the Appium organization in GitHub

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Appium Server crashed because of `javascript heap out of memory` in 350 time loop `driver.screenshot`

### Expected Behavior

Appium Server can normally screenshot during loop

### Minimal Reproducible Example

```
while True:
    driver.screenshot('1.png')
```

### Environment

- Operating system:
- Appium server version (output of `appium --version`): 2.0.0
- Appium driver(s) and their version(s) (`appium driver list`):  uiautomator2@2.29.11 [installed (npm)]
- Appium plugin(s) and their version(s) (`appium plugin list`): None
- Node.js version (output of `node --version`): v20.11.1
- `npm` version (output of `npm --version`): 10.2.4
- Last component(s) version which did _not_ exhibit the problem:
- Platform and version under test: Ubuntu 22.04
- Real device or emulator/simulator: real device

### Further Information

I dump the heap snapshot  and  find the `_global_npmlog` don't release memory normally
<img width="1437" alt="Screenshot 2024-08-12 at 11 15 26" src="https://github.com/user-attachments/assets/fd8c93a7-a05a-4fdf-9343-bf392d77feab">

Due to security concerns, I cannot directly upload the heap file. If there is any other information that needs to be provided, please let me know.

**Comment by mykola-mokhnach** [comment]
Please make sure you use the most recent driver and server versions.

**Comment by lanbaoshen** [comment]
> Please make sure you use the most recent driver and server versions.

I found code bellow:
```
 // otherwise, either use the global, or a new `npmlog` object
 logger = global._global_npmlog || npmlog;
 // The default value is 10000, which causes excessive memory usage
 logger.maxRecordSize = MAX_LOG_RECORDS_COUNT; 
```
I think if my single image is very large, it would exceed the default memory limit before reaching the `MAX_LOG_RECORDS_COUNT` log entries cap.

**Comment by mykola-mokhnach** [comment]
Unfortunately we do not patch older versions, and cannot help there much

**Comment by lanbaoshen** [comment]
> Unfortunately we do not patch older versions, and cannot help there much

Yeah, I update appium server to 2.11.3, and  check heap snapshot again, I've found that each member of the array has been reduced by about 3000 times, perhaps this issue has already been fixed. 

I think I won't encounter an Out of Memory (OOM) situation in this scenario anymore. 

<img width="1255" alt="Screenshot 2024-08-12 at 15 29 44" src="https://github.com/user-attachments/assets/e9b61f9b-2f1f-42dc-a1e7-bdb9bc537578">

Thank you for your help!
